hvac/api/auth_methods/kubernetes.py

```diff
@@ -112,7 +113,8 @@ def create_role(
         self,
         name,
         bound_service_account_names,
-        bound_service_account_namespaces,
+        bound_service_account_namespaces=None,
+        bound_service_account_namespace_selector=None,
```

Comment:
Right now, we require any new parameters to be at the end of the list, so that it doesn't break clients that use positional parameters (I'd like to eventually make most parameters in the library kw-only but that's a large undertaking).

We'll also need to account for `bound_service_account_namespaces` becoming optional now, I'd like to see the tests reflect that change by testing both no value passed as well as an explicit value.

Reply:
I added an integration testcase where no `bound_service_account_namespaces` is empty. Do you need another test?

For the other concern I can restructure the parameters to append them at the end. Can you give me an example inside this repo to help me understand what the end result should look like?

Reply:
No need to worry about kw-only right now.

For the test that might be ok yah, I don't see the CI run anymore and for some reason no coverage report was posted so let's see how that shakes out once the other change is pushed.

Reply:
It seems like there are just doctests running.
When testing locally I get CA Errors. This seems not to be
```
FAILED tests/integration_tests/api/auth_methods/test_kubernetes.py::TestKubernetes::test_configure_5_missing_kubernetes_ca_cert - AssertionError: InvalidRequest not raised
```
and on test_list_roles I get:
```
FAILED tests/integration_tests/api/auth_methods/test_kubernetes.py::TestKubernetes::test_list_roles_0_success - hvac.exceptions.InvalidRequest:  The provided CA PEM data contains no valid certificates, on post https://127.0.0.1:48195/v1/auth/kubernetes-test/config
```

I can push another change when needed to trigger the CI again.

Reply:
sorry I missed that last push. I've re-run CI. Looks like you've got some lint/formatting errors and more importantly the integration tests are failing, new error though, no certificate errors as far as I can tell.

I recommend running formatting and tests locally to give yourself a better feedback loop.